tests/CMakeLists.txt

```diff
@@ -36,23 +36,35 @@ MACRO(ADD_TESTCASE NAME GENERATED)
   IF (${GENERATED} STREQUAL TRUE)
     ADD_UNIT_TEST(${NAME} ${CMAKE_CURRENT_BINARY_DIR}/${NAME}.cc)
   ELSE()
-    ADD_UNIT_TEST(${NAME} ${NAME}.cc)
+    IF(BUILD_UNIT_TESTS)
+      ADD_UNIT_TEST(${NAME} ${NAME}.cc)
+    ELSE(BUILD_UNIT_TESTS)
+      ADD_UNIT_TEST(${NAME} ${NAME}.cc EXCLUDE_FROM_ALL ${NAME})
+    ENDIF(BUILD_UNIT_TESTS)
   ENDIF()
 
   # Link against Boost and project library.
   TARGET_LINK_LIBRARIES(${NAME}
     ${Boost_LIBRARIES}
     ${PROJECT_NAME})
-  PKG_CONFIG_USE_DEPENDENCY(${NAME} assimp)
   PKG_CONFIG_USE_DEPENDENCY(${NAME} hpp-util)
-  PKG_CONFIG_USE_DEPENDENCY(${NAME} hpp-fcl)
   PKG_CONFIG_USE_DEPENDENCY(${NAME} pinocchio)
+
+  # Add test to check command
+  ADD_DEPENDENCIES(check ${NAME})
 ENDMACRO(ADD_TESTCASE)
 
 MACRO(ADD_TESTCASE_CFLAGS target flag)
   SET_PROPERTY(TARGET ${target} APPEND_STRING PROPERTY COMPILE_FLAGS " ${flag}")
 ENDMACRO(ADD_TESTCASE_CFLAGS)
 
+# Define check command (just run make check to compile and run all the checks)
+IF(APPLE)
+  ADD_CUSTOM_TARGET(check COMMAND export DYLD_LIBRARY_PATH=$ENV{DYLD_LIBRARY_PATH} && ${CMAKE_CTEST_COMMAND})
```

Comment:
On apple, it is necessary to manually pass the DYLD_LIBRARY_PATH ?

Can this be done in the submodule in `ADD_UNIT_TEST` using `SET_TESTS_PROPERTIES(test1 [test2 ...] PROPERTIES ENVIRONMENT "DYLD_LIBRARY_PATH=$ENV{DYLD_LIBRARY_PATH}")` ?

Reply:
I was not using the jrl-cmakemodules for testing. I will have a look and check all these points.

Reply:
Thanks